src/Mod/Part/Gui/TaskAttacher.cpp

```diff
@@ -431,17 +564,43 @@ void TaskAttacher::onSelectionChanged(const Gui::SelectionChanges& msg)
         }
 
         if (autoNext) {
-            if (iActiveRef == -1){
+            if (iActiveRef == -1) {
                 //nothing to do
-            } else if (iActiveRef == 4 || this->lastSuggestResult.nextRefTypeHint.empty()){
+            }
+            else if (iActiveRef == 4 || this->lastSuggestResult.nextRefTypeHint.empty()) {
                 iActiveRef = -1;
-            } else {
+            }
+            else {
                 iActiveRef++;
             }
         }
+    }
 
-        updateReferencesUI();
+    try {
+        updateListOfModes();
+        eMapMode mmode = getActiveMapMode();  //will be mmDeactivated, if selected or if no modes are available
+        if (mmode == mmDeactivated) {
+            //error = true;
+            this->completed = false;
+        }
+        else {
+            this->completed = true;
+        }
+        pcAttach->MapMode.setValue(mmode);
+        selectMapMode(mmode);
+        updatePreview();
     }
+    catch (Base::Exception& e) {
+        ui->message->setText(QCoreApplication::translate("Exception", e.what()));
+        ui->message->setStyleSheet(QString::fromLatin1("QLabel{color: red;}"));
+    }
+
+    updateReferencesUI();
+}
+
+void TaskAttacher::addToReference(SubAndObjName pair)
+{
+    addToReference({ pair });

```

Comment:
This will cause a crash with an infinite recursion because it won't call the overloaded function `addToReference(const std::vector<SubAndObjName>& pairs)` but again `addToReference(SubAndObjName pair)`.

To make sure it calls the desired function you must write `addToReference({{{ pair }}});`.

Test application to demonstrate what happens:
```
#include <string>
#include <vector>
#include <iostream>

struct SubAndObjName {
    std::string objName;
    std::string subName;
};


void test(const std::vector<SubAndObjName>& v)
{
    std::cout << "test(const std::vector<SubAndObjName>&) with " << v.size() << " elements\n";
}

void test(SubAndObjName pair)
{
    std::cout << "test(SubAndObjName>)\n";
}

int main()
{
    SubAndObjName pair;
    test({pair});
    test({{{pair}}});
}
```
The output will be:
```
test(SubAndObjName>)
test(const std::vector<SubAndObjName>&) with 1 elements
```

Reply:
Thanks I'll fix it tomorrow morning (unless you want to fix it before)